src/ecs/explorer/ecsServiceNode.ts

```diff
@@ -39,8 +40,15 @@ export class EcsServiceNode extends AWSTreeNodeBase implements AWSResourceNode {
         return await makeChildrenNodes({
             getChildNodes: async () => {
                 const containerNames = await this.ecs.getContainerNames(this.service.taskDefinition!)
-                return containerNames.map(
-                    name => new EcsContainerNode(name, this.name, this.parent.arn, this.ecs, this)
+                return await Promise.all(
+                    containerNames.map(async name => {
+                        const tasks = await this.ecs.listTasks(
```

Comment:
Does the Promise.all potentially block the rendering of this parent node? Why not calculate the tasks in the child node? That also means, in the future, when we have selective refresh of nodes, the nodes will recalculate as needed. Whereas with this code, we would need to refresh the parent in order to refresh the children.

The need for a `hasRunningTasks` property was a hint that the calculation could be shifted closer to its usage.

Reply:
`getChildren` will block the entire tree from rendering. Tree doesn't render until the model (all nodes and their children) are resolved.

To me, this code belongs in the child node if only because we are trying to get resources using the node. The service node shouldn't need to care unless it's explicitly taking ownership of all tasks.